### PR TITLE
Fix issues in RDF model output concerning property constraints and vocab instances

### PR DESCRIPTION
**Pull request by fholger**
While experimenting with json-ld serialization and shacl validation, we noticed a couple of errors in the generated RDF module. This PR aims to fix the following issues:

- instances of vocabulary types were generated with the wrong URI, which was missing the vocabulary type as a prefix
- `sh:datatype` was used as a constraint for all class properties, but it only works for literal types. For our complex types, `sh:class` needs to be used, instead. (In some instances, like `PositiveIntegerRange`, `sh:node` might be better / more convenient, but `sh:class` at least works.)
- `spdxId` was included as a required property, even though RDF already has the `@id` field that should be used instead.

### CODE REVIEW

File: spec_parser/utils.py [hunk 1]
```diff
@@ -658,5 +681,6 @@ def _gen_rdf(self, g: rdflib.Graph):
 
         # add entries
         for _entry, _desc in self.entries.items():
-            g.add((self._gen_uri(_entry), RDF.type, OWL.NamedIndividual))
-            g.add((self._gen_uri(_entry), RDF.type, cur))
+            uri = cur + "/" + _entry
+            g.add((uri, RDF.type, OWL.NamedIndividual))
+            g.add((uri, RDF.type, cur))
```
Comment (armintaenzertng) [inline spec_parser/utils.py:686]: Why did you change this?

**Review by armintaenzertng — APPROVED** [review]
Thanks! I have two non-blocking remarks.

File: spec_parser/utils.py (hunk 1 above)
Comment (fholger) [reply spec_parser/utils.py:686]: That's the fix for the first issue - `_gen_uri` generates an inccorect URIRef for the vocab elements that's missing the vocab type prefix.

File: spec_parser/utils.py [hunk 2]
```diff
@@ -435,15 +447,23 @@ def _gen_rdf(self, g: rdflib.Graph) -> None:
         g.add((cur, RDFS.comment, Literal(self.description)))
         g.add((cur, NS0.term_status, Literal(self.metadata.get("Status")[0])))
 
+        sh_class = URIRef("http://www.w3.org/ns/shacl#class")
```
[the remark this answers is not in the document]
Comment (armintaenzertng) [reply spec_parser/utils.py:450]: Weird, but ok :)

**Review by armintaenzertng — APPROVED** [review]

**Review by zvr — APPROVED** [review]
